src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/GoogleDataprocDAO.scala

```diff
@@ -349,6 +330,80 @@ class GoogleDataprocDAO(protected val leoServiceAccountEmail: WorkbenchEmail,
     }
   }
 
+  /* Create the staging bucket for the dataproc cluster*/
+  override def createStagingBucket(bucketGoogleProject: GoogleProject, clusterGoogleProject: GoogleProject, stagingBucketName: GcsBucketName, serviceAccountInfo: ServiceAccountInfo, groupStagingAcl: List[WorkbenchEmail], userStagingAcl: List[WorkbenchEmail]): Future[GcsBucketName] = {
+
+    // The Leo service account
+    val leoServiceAccountEntityString = s"user-${leoServiceAccountEmail.value}"
+
+    val clusterServiceAccountEntityStringFuture: Future[String] = getClusterServiceAccountEntity(clusterGoogleProject, serviceAccountInfo)
+
+    clusterServiceAccountEntityStringFuture.flatMap { clusterServiceAccountEntityString =>
+      //Add the Leo SA and the cluster's SA to the ACL list for the bucket
+      val bucketAcls = List(
+        createBucketAcl(clusterServiceAccountEntityString,"READER"),
+        createBucketAcl(leoServiceAccountEntityString,"OWNER")
+      ) ++ groupStagingAcl.map(a => createBucketAcl("group-" + a, "READER")) ++
```

Comment:
Hm, `GoogleStorageDAO` right now can't handle group ACLs, only user ACLs. Might need to add that.

Reply:
+1

Reply:
Done